workspaces/api-gateway/src/api/v1/webhooks/webhooks.controller.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Controller,
+  InternalServerErrorException,
+  Post,
+  RawBodyRequest,
+  Req,
+} from '@nestjs/common';
+import Stripe from 'stripe';
+import { InjectRepository } from '@nestjs/typeorm';
+import type { Repository } from 'typeorm';
+import type { Request } from 'express';
+
+import { Payment } from 'database/models/payment.entity';
+import { stripe } from 'database/models/paymentMethods/stripeMethod.entity';
+import { convertByOption } from 'utils';
+import type { IMetadataParams } from 'api/v1/payments/payments.interface';
+
+@Controller('webhook')
+export class WebhooksController {
+  private stripe: Stripe;
+  private endpointSecret: string;
+
+  constructor(
+    @InjectRepository(Payment)
+    private paymentRepository: Repository<Payment>,
+  ) {
+    this.stripe = stripe;
+    this.endpointSecret = process.env.SECRET_ENDPOINT_STRIPE;
+  }
+
+  @Post('/stripe')
+  public async createCheckoutStripe(
+    @Req() request: RawBodyRequest<Request>,
+  ): Promise<Stripe.Event> {
+    const payload = request.rawBody;
+    const sig = request.headers['stripe-signature'];
+    let event: Stripe.Event;
+
+    try {
+      event = this.stripe.webhooks.constructEvent(
+        payload,
+        sig,
+        this.endpointSecret,
+      );
+    } catch (err) {
+      throw new InternalServerErrorException(err);
+    }
+
+    const session = convertByOption<Stripe.Event.Data.Object>(
+      'camelCase',
+      event.data.object,
+    );
+
+    switch (event.type) {
+      case 'charge.succeeded': {
+        const metadata: IMetadataParams = {
+          email: session['billingDetails']['email'],
+          brandCard: session['paymentMethodDetails']['card']['brand'],
+          country: session['paymentMethodDetails']['card']['country'],
+          expMonth: session['paymentMethodDetails']['card']['expMonth'],
+          expYear: session['paymentMethodDetails']['card']['expYear'],
+          last4: session['paymentMethodDetails']['card']['last4'],
```

Comment:
why don't we use `session.billingDetails`?

Reply:
Em thử rồi nhưng nó không hiện ra attribute nào hết ạ

Reply:
check the typing

Reply:
Em rê chuột vào session thì type của nó hiện là Stripe.Event.Data.Object